src/frontend/komponenter/Fagsak/Personlinje/Behandlingsmeny/LeggBehandlingPåVent/SettBehandlingPåVentModal.tsx

```diff
@@ -25,82 +26,101 @@ const StyledBodyShort = styled(BodyShort)`
 `;
 
 interface IProps {
-    visModal: boolean;
-    onAvbryt: () => void;
-    settBehandlingPåVent: () => void;
-    skjema: ISkjema<{ frist: Date | undefined; årsak: SettPåVentÅrsak | undefined }, IBehandling>;
-    erBehandlingAlleredePåVent: boolean;
+    lukkModal: () => void;
+    behandling: IBehandling;
 }
 
-export const SettBehandlingPåVentModal: React.FC<IProps> = ({
-    visModal,
-    onAvbryt,
-    settBehandlingPåVent,
-    skjema,
-    erBehandlingAlleredePåVent,
-}) => {
+export const SettBehandlingPåVentModal: React.FC<IProps> = ({ lukkModal, behandling }) => {
     const årsaker = hentAlleÅrsaker();
+    const { skjema, kanSendeSkjema, onSubmit } = useSettPåVentSkjema(behandling.behandlingPåVent);
+    const { settÅpenBehandling } = useBehandling();
+
+    const erBehandlingAlleredePåVent = !!behandling.behandlingPåVent;
+
+    const url = erBehandlingAlleredePåVent
+        ? `/familie-ks-sak/api/behandlinger/${behandling.behandlingId}/sett-på-vent/oppdater`
+        : `/familie-ks-sak/api/behandlinger/${behandling.behandlingId}/sett-på-vent`;
+
+    const settBehandlingPåVent = () => {
+        if (kanSendeSkjema()) {
+            onSubmit(
+                {
+                    method: erBehandlingAlleredePåVent ? 'PUT' : 'POST',
+                    data: { frist: skjema.felter.frist.verdi, årsak: skjema.felter.årsak.verdi },
+                    url: url,
+                },
+                (ressurs: Ressurs<IBehandling>) => {
+                    settÅpenBehandling(ressurs);
+                    lukkModal();
+                }
+            );
+        }
+    };
```

Comment:
Ser at det skjedde litt mye i denne committen, men jeg flyttet også inn alt av data i denne filen i stedet for å sende det med fra SettEtterOppdaterVenting.tsx fordi det kun ble brukt her